fix(workflows): Minor fixes affecting release pipelines

nderjung opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

nderjung committed:
fix(workflows): Add missing OpenSSH client for AUR release

Signed-off-by: Alexander Jung <alex@unikraft.io>

nderjung committed:
fix(workflows): Fetch Git tags for KraftKit versioning

Signed-off-by: Alexander Jung <alex@unikraft.io>

craciunoiuc reviewed (approved):
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

nderjung committed:
fix(workflows): Use "Path context" in OCI build step

This forces [0] the working directory instead of using a Git-based
context, which will allow the the `Dockerfile`'s internal step
which runs `make` to correctly seed Git (and version) information.

[0]: https://github.com/docker/build-push-action/tree/v6.14.0?tab=readme-ov-file#git-context

Signed-off-by: Alexander Jung <alex@unikraft.io>

nderjung committed:
fix(Makefile): Use `--always` to force Git description

Signed-off-by: Alexander Jung <alex@unikraft.io>
